### PR TITLE
(PROPERLY) Installs (official) Origin if it's not present

### PR DESCRIPTION
**Pull request by VictorPLopes**
New PR because I changed the download source, since you wanted everything to use official links... I still think it's a bad idea to download and then downgrade it though.

### CODE REVIEW

File: Program.cs [hunk 1]
```diff
@@ -4,37 +4,47 @@
 using DyviniaUtils;
 using Microsoft.Win32;
 
-namespace StayOnOrigin {
-    internal class Program {
+namespace StayOnOrigin
+{
+    internal class Program
+    {
```
Comment (Dyvinia) [inline Program.cs]: fix pls

File: Program.cs [hunk 2]
```diff
@@ -4,37 +4,47 @@
...
         public static string OriginPath => Registry.LocalMachine.OpenSubKey(@"SOFTWARE\WOW6432Node\Origin")?.GetValue("OriginPath")?.ToString();
         public static Version OriginVersion => new(FileVersionInfo.GetVersionInfo(OriginPath).FileVersion.Replace(",", "."));
         public static string TempDirPath => Path.Combine(Environment.CurrentDirectory, "temp");
 
-        static void Main() {
+        static void Main()
+        {
```
Comment (Dyvinia) [inline Program.cs]: fix pls

File: Program.cs [hunk 3]
```diff
@@ -4,37 +4,47 @@
...
+                WriteSeparator();
             }
 
+            // Kill All Origin/EA related processes
+            KillProcesses();
+
             // Check if Origin is too new (Anything after 10.5.120.x)
-            if (OriginVersion.CompareTo(new("10.5.120.0")) > 0) {
+            if (OriginVersion.CompareTo(new("10.5.120.0")) > 0)
+            {
```
Comment (Dyvinia) [inline Program.cs]: fix pls

File: Program.cs [hunk 4]
```diff
@@ -4,37 +4,47 @@
...
                 Console.WriteLine($"Origin v{OriginVersion} is too recent");
                 Console.WriteLine("Downgrading to Origin v10.5.118.52644...");
                 ResetTempDir();
                 UpdateOrigin().Wait();
                 WriteSeparator();
             }
-            else {
+            else
+            {
```
Comment (Dyvinia) [inline Program.cs]: fix pls

File: Program.cs [hunk 5]
```diff
@@ -55,7 +65,32 @@ internal class Program {
             Console.ReadKey();
         }
 
-        static async Task UpdateOrigin() {
+        static async Task InstallOrigin()
+        {
```
Comment (Dyvinia) [inline Program.cs]: fix pls

File: Program.cs [hunk 6]
```diff
@@ -55,7 +65,32 @@ internal class Program {
...
+            // Install
+            var originInstall = Process.Start(destinationPath);
+            Console.WriteLine("Origin is being installed...");
+            originInstall.WaitForExit();
+            Console.WriteLine();
+            Console.WriteLine($"Installed Origin");
+        }
+
+        static async Task UpdateOrigin()
+        {
```
Comment (Dyvinia) [inline Program.cs]: fix pls

File: Program.cs [hunk 7]
```diff
@@ -72,17 +107,20 @@ internal class Program {
             // Install
             using ZipArchive archive = ZipFile.OpenRead(destinationPath);
             int i = 0;
-            foreach (ZipArchiveEntry entry in archive.Entries) {
-                int percentage = Convert.ToInt32(100*i++/(float)archive.Entries.Count);
+            foreach (ZipArchiveEntry entry in archive.Entries)
+            {
```
Comment (Dyvinia) [inline Program.cs]: fix pls

File: Program.cs [hunk 8]
```diff
@@ -72,17 +107,20 @@ internal class Program {
...
+                int percentage = Convert.ToInt32(100 * i++ / (float)archive.Entries.Count);
```
Comment (Dyvinia) [inline Program.cs:104]: (keep this)

File: Program.cs [hunk 9]
```diff
@@ -72,17 +107,20 @@ internal class Program {
...
             Console.WriteLine();
             Console.WriteLine($"Installed Origin v10.5.118.52644");
         }
 
-        static void ClearFile(string path, bool readOnly) {
-            if (File.Exists(path) && !File.GetAttributes(path).HasFlag(FileAttributes.ReadOnly)) {
+        static void ClearFile(string path, bool readOnly)
+        {
+            if (File.Exists(path) && !File.GetAttributes(path).HasFlag(FileAttributes.ReadOnly))
+            {
```
Comment (Dyvinia) [inline Program.cs]: fix pls

File: Program.cs [hunk 10]
```diff
@@ -97,7 +135,8 @@ internal class Program {
             }
         }
 
-        static void DisableMigration() {
+        static void DisableMigration()
+        {
```
Comment (Dyvinia) [inline Program.cs]: fix pls

File: Program.cs [hunk 11]
```diff
@@ -111,7 +150,8 @@ internal class Program {
                 "AutoUpdate",
                 "/Settings"
             };
-            foreach (string line in fileLines) {
+            foreach (string line in fileLines)
+            {
```
Comment (Dyvinia) [inline Program.cs]: fix pls

File: Program.cs [hunk 12]
```diff
@@ -128,19 +168,23 @@ internal class Program {
             File.WriteAllLines(localXML, fileLinesNew.ToArray());
         }
 
-        static void KillProcesses() {
+        static void KillProcesses()
+        {
```
Comment (Dyvinia) [inline Program.cs]: fix pls

File: Program.cs [hunk 13]
```diff
@@ -128,19 +168,23 @@ internal class Program {
...
             bool addSeparator = false;
-            foreach (Process process in Process.GetProcessesByName("EADesktop")) {
+            foreach (Process process in Process.GetProcessesByName("EADesktop"))
+            {
```
Comment (Dyvinia) [inline Program.cs]: fix pls

File: Program.cs [hunk 14]
```diff
@@ -128,19 +168,23 @@ internal class Program {
...
                 process.Kill();
                 Console.WriteLine($"Killed {process.ProcessName}");
                 addSeparator = true;
             }
-            foreach (Process process in Process.GetProcessesByName("Origin")) {
+            foreach (Process process in Process.GetProcessesByName("Origin"))
+            {
```
Comment (Dyvinia) [inline Program.cs]: fix pls

File: Program.cs [hunk 15]
```diff
@@ -128,19 +168,23 @@ internal class Program {
...
                 process.Kill();
                 Console.WriteLine($"Killed {process.ProcessName}");
                 addSeparator = true;
             }
-            foreach (Process process in Process.GetProcessesByName("OriginWebHelperService")) {
+            foreach (Process process in Process.GetProcessesByName("OriginWebHelperService"))
+            {
```
Comment (Dyvinia) [inline Program.cs]: fix pls

File: Program.cs [hunk 16]
```diff
@@ -149,15 +193,17 @@ internal class Program {
                 WriteSeparator(); // Prevent double separator from appearing in console
         }
 
-        static void ResetTempDir(bool recreateDir = true) {
+        static void ResetTempDir(bool recreateDir = true)
+        {
```
Comment (Dyvinia) [inline Program.cs]: fix pls

File: Program.cs [hunk 17]
```diff
@@ -149,15 +193,17 @@ internal class Program {
...
             if (Directory.Exists(TempDirPath))
                 Directory.Delete(TempDirPath, true);
             if (recreateDir)
                 Directory.CreateDirectory(TempDirPath);
         }
 
-        static void WriteSeparator() {
+        static void WriteSeparator()
+        {
```
Comment (Dyvinia) [inline Program.cs]: fix pls

**Review by Dyvinia — CHANGES_REQUESTED** [review]

**Review by VictorPLopes — COMMENTED** [review]
Fixed. Thank you for the heads up.

**Pull request merged**